### PR TITLE
Add support to import a dex client resource, update dexApi

### PR DESCRIPTION
**Pull request by koendelaat**
Based on #7 
New DexIdP release: https://github.com/dexidp/dex/releases/tag/v2.38.0

### CODE REVIEW

**Comment by marcofranssen** [comment]
https://github.com/dexidp/dex/pull/2972#issuecomment-1975948535 It seems it isn't officially released yet on the Go API.

File: pkg/dexidp/dexclient_resource_test.go [hunk 1]
```diff
@@ -34,6 +34,15 @@ resource "dexidp_client" "test_client" {
...
 				),
 			},
+			// ImportState testing
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				// The last_updated attribute does not exist in the Dex gRPC
+				// API, therefore there is no value for it during import.
+				ImportStateVerifyIgnore: []string{"last_updated"},
```
Comment (marcofranssen) [inline pkg/dexidp/dexclient_resource_test.go:44]: Maybe we should just set the date terraform imported it. I think the other parts we also use the date, terraform created/updated the resource.

File: go.sum [hunk 2]
```diff
@@ -160,8 +158,10 @@ github.com/mitchellh/mapstructure v1.5.0/go.mod h1:bFUtVrKA4DC2yAKiSyO/QUcy7e+RR
 github.com/mitchellh/reflectwalk v1.0.0/go.mod h1:mSTlrgnPZtwu0c4WaC2kGObEpuNDbx0jmZXqmk4esnw=
 github.com/mitchellh/reflectwalk v1.0.2 h1:G2LzWKi524PWgd3mLHV8Y5k7s6XUvT0Gef6zxSIeXaQ=
 github.com/mitchellh/reflectwalk v1.0.2/go.mod h1:mSTlrgnPZtwu0c4WaC2kGObEpuNDbx0jmZXqmk4esnw=
-github.com/oklog/run v1.0.0 h1:Ru7dDtJNOyC66gQ5dQmaCa0qIsAUFY3sFpK1Xk8igrw=
-github.com/oklog/run v1.0.0/go.mod h1:dlhp/R75TPv97u0XWUtDeV/lRKWPKSdTuV0TZvrmrQA=
+github.com/oklog/run v1.1.0 h1:GEenZ1cK0+q0+wsJew9qUg/DyD8k3JzYsZAi5gYi2mA=
+github.com/oklog/run v1.1.0/go.mod h1:sVPdnTZT1zYwAJeCMu2Th4T21pA3FPOQRfWjQlk7DVU=
+github.com/philips-forks/dex/api/v2 v2.1.1-0.20240826130519-d7839e1fa7b1 h1:nFGgvtcebYXd4yQmJNsJ1UmWy9FiVQ//MPohafgcdKU=
```
Comment (marcofranssen) [inline go.sum:163]: Don't want to base this provider on sources from a Fork. We need to work with upstream project to adopt the features from that fork.

File: go.mod [hunk 3]
```diff
@@ -76,17 +76,19 @@ require (
...
-	golang.org/x/net v0.25.0 // indirect
+	golang.org/x/net v0.27.0 // indirect
 	golang.org/x/sys v0.23.0 // indirect
 	golang.org/x/text v0.17.0 // indirect
 	google.golang.org/appengine v1.6.8 // indirect
-	google.golang.org/protobuf v1.34.1 // indirect
+	google.golang.org/protobuf v1.34.2 // indirect
 )
+
+replace github.com/dexidp/dex/api/v2 v2.1.0 => github.com/philips-forks/dex/api/v2 v2.1.1-0.20240826130519-d7839e1fa7b1
```
Comment (marcofranssen) [inline go.mod:94]: Don't want to base this provider on sources from a Fork. We need to work with upstream project to adopt the features from that fork.

File: go.sum (hunk 2 above)
Comment (koendelaat) [reply go.sum:163]: Upstream PR: https://github.com/dexidp/dex/pull/3598

File: go.mod (hunk 3 above)
Comment (koendelaat) [reply go.mod:94]: Upstream PR: https://github.com/dexidp/dex/pull/3598

**Comment by marcofranssen** [comment]
AutoMerged the original #7 earlier when I bumped dependencies and pushed those. I see still few small changes here in yours. Do you feel those changes are required, regarding defaults?

**Pull request closed without merging**